nemo/collections/asr/parts/mixins/transcription.py

```diff
@@ -0,0 +1,542 @@
+
+                    # clear up memory
+                    del test_batch, model_outputs
+
+                    # Yield results if generator
+                    yield processed_outputs
+
+                    del processed_outputs
+
+        finally:
+            # set mode back to its original value
+            self._transcribe_on_end(transcribe_cfg)
+
+    """
+    Transcribe Execution Flow
+    """
+
+    def _transcribe_on_begin(self, audio, trcfg: TranscribeConfig):
+        if audio is None:
+            return {}
+
+        if isinstance(audio, (str, np.ndarray, torch.Tensor)):
+            audio = [audio]
+
+        if isinstance(audio, list) and len(audio) == 0:
+            return {}
+
+        if trcfg._internal.device is None:
+            trcfg._internal.device = next(self.parameters()).device
+
+        if trcfg._internal.dtype is None:
+            trcfg._internal.dtype = next(self.parameters()).dtype
+
+        if trcfg.num_workers is None:
+            trcfg.num_workers = min(trcfg.batch_size, os.cpu_count() - 1)
+
+        # Model's mode and device
+        trcfg._internal.training_mode = self.training
+
+        # Switch model to evaluation mode
+        if hasattr(self, 'preprocessor'):
+            if hasattr(self.preprocessor, 'featurizer') and hasattr(self.preprocessor.featurizer, 'dither'):
+                trcfg._internal.dither_value = self.preprocessor.featurizer.dither
+                self.preprocessor.featurizer.dither = 0.0
+
+            if hasattr(self.preprocessor, 'featurizer') and hasattr(self.preprocessor.featurizer, 'pad_to'):
+                trcfg._internal.pad_to_value = self.preprocessor.featurizer.pad_to
+                self.preprocessor.featurizer.pad_to = 0
+
+        # Switch model to evaluation mode
+        self.eval()
+
+        # Disable logging
+        trcfg._internal.logging_level = logging.get_verbosity()
+        logging.set_verbosity(logging.WARNING)
+
+    def _transcribe_input_processing(self, audio, trcfg: TranscribeConfig):
+        if isinstance(audio, (list, tuple)):
+            if len(audio) == 0:
+                raise ValueError("Input `audio` is empty")
+        else:
+            audio = [audio]
+
+        if isinstance(audio[0], str):
+            audio_files = list(audio)
+
+            tmp_dir = trcfg._internal.temp_dir
+            ds_config = self._transcribe_input_manifest_processing(audio_files, tmp_dir, trcfg)
+
+            temp_dataloader = self._setup_transcribe_dataloader(ds_config)
+            return temp_dataloader
+
+        elif isinstance(audio[0], (np.ndarray, torch.Tensor)):
+            audio_tensors = list(audio)
+
+            # Convert numpy tensors to torch tensors
+            if any([isinstance(_tensor, np.ndarray) for _tensor in audio_tensors]):
+                audio_tensors = [
+                    torch.from_numpy(audio_tensor) if isinstance(audio_tensor, np.ndarray) else audio_tensor
```

Comment:
you can simplify with [`torch.as_tensor`](https://pytorch.org/docs/stable/generated/torch.as_tensor.html)

Reply:
Done. Ill keep the check for np as apparabtly it will duplicate memory if it finds same tensor but different dtype